src/tnetdbg.hpp

```diff
@@ -173,6 +173,23 @@ auto constexpr debug_config_erro_backtrace_level = 128; ///< the backtrace level
 		<< "."); \
 		throw except_var; } while(0)
 
+template <class T>
+void throw_or_abort(const T & ex) {
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wterminate"
+#endif
+	throw ex;
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic pop
+#endif
+}
+
+#define _throw_error_or_abort( EXCEPT ) do { auto except_var = (EXCEPT);  \
```

Comment:
/***
 * This function tries to throw, BUT ALSO it is O.K. with possible ABORT, due to noexcept on the function we are in
 * it exists to e.g. silence some compilers warning about possible throw inside noexcept - we know, and we are o.k. with it turning into abort
 * use it in macros that should try to throw but abort if throwing is not possible
*/